Develop

alannaesbenshade opened the pull request:
added sass variables and mixins.

alannaesbenshade committed:
added project files

alannaesbenshade committed:
css

fixed css file ref

alannaesbenshade committed:
Sass

Updated sass- nested css and created a few variables

alannaesbenshade committed:
variable

added variable for roboto

alannaesbenshade committed:
mixin

added link mixin

alannaesbenshade committed:
breadcrumb

added links to bread crumb (#)

philschanely commented:
Reviewing. Will be in touch.

philschanely commented on sass/styles.scss:
```diff
@@ -1 +1,547 @@
+// $active: #000000;
+
+/*mixins*/
+@mixin link($link, $hover, $active) {
```
Nice mixin idea here!

philschanely replied:
But now seeing its use it might be a bit fabricated. A better use for mixin is for code you repeat frequently. I'll look for such instances and flag them.

philschanely commented on sass/styles.scss:
```diff
@@ -1 +1,547 @@
+  }
+}
+
+$html-margin: 0px auto !default;
```
Not sure `!default` is necessary here, but good to know its available in Sass.

philschanely commented on sass/styles.scss:
```diff
@@ -1 +1,547 @@
+    display:none;
+}
+
+h2 {
```
Watch for and remove redundancy/repetition. For example, this setting appears here and in two other media queries with no changes. Reduce to the one necessary setting. Look for other similar situations and remove them.

philschanely replied:
Hint: Use the Find feature in Atom to make this quick and easy to look through your code for same selectors and compare to reduce repetition.

philschanely commented on sass/styles.scss:
```diff
@@ -1 +1,547 @@
+      margin-bottom:20px;
+  }
+
+  .products ul {
```
Just checking here.. do you have `.products ul` within `.products` or is this a nesting mistake?

alannaesbenshade replied:
that was a mistake- I fixed it

philschanely commented on sass/styles.scss:
```diff
@@ -1 +1,547 @@
+      display:block;
+      height:25px;
+  }
+  &:after {
```
here's a good example of a good candidate for a mixin. You use this `&:after...` clearfix 5 times as written. Consider packaging this into a `clearfix` mixin then use `@include` to replace each instance of this code.

philschanely commented on sass/styles.scss:
```diff
@@ -1 +1,547 @@
-p {color:red;}
+@import url('https://fonts.googleapis.com/css?family=Roboto+Condensed:300,400,700|Roboto:100,300,400,500,700');
```
It would be nice to see this large file broken out into partials as demonstrated in my videos and described [here](http://sass-lang.com/guide#topic-4). This way you can group type settings together, layout settings together, etc. Or you could use them to piece out media queries so you can compare them side-by-side rather than having to scroll up and down a lot.

philschanely commented on sass/styles.scss:
```diff
@@ -1 +1,547 @@
+.review {
+  width:320px;
+  display:block;
+  min-height:200px;
```
While this `min-height` helps to enforce a grid perhaps it leads to undesired extra space in the shorter review. Consider removing this.

philschanely replied:
Looking again the problem may instead by the `h3` you've nested in the first item. I'll make a comment on that in the markup.

philschanely commented on detail.html:
```diff
@@ -0,0 +1,144 @@
+
+<ul>
+    <li class="review">
+            <h3>Reviews</h3>
```
This is a bit of a shortcut hack way to get this item aligned with the other reviews and it ends up bumping the content in this `.review` down making the spacing awkward. I've made a comment about setting the `min-height` but take that into consideration long with moving this `h3` out of the list and over the `ul` as a whole. This is a more semantic appropriate thing given that this heading really labels the whole list. Then you can wrap the `h3` and `ul` in a `div` and move the `width`, `margin`, etc. from `.review` items to the new grouping `div`. This will ensure the list and `h3` are aligned.

philschanely commented on index.html:
```diff
@@ -0,0 +1,131 @@
+        <ul>
+
+            <li>
+                <img class="product-img" src="images/img1.png" />
```
As with the original submission of this project, nice microformatting here on these items!

philschanely reviewed (changes requested):
Good start overall. I have some picky comments to help you take this further. Please make some revisions as requested, commit them, and let me know via comment when you're ready for further review.

alannaesbenshade committed:
Updated

Fixed (hopefully) issues brought up in the review comments- I have the typeface imported from a partial file, I made a mixin for the clearfix, and I trimmed some redundant code. I also restructured a few minor elements.
